mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceProperties.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration properties for the Mail Source module.
+ *
+ * @author Amol
+ */
+@ConfigurationProperties
+public class MailSourceProperties {
+
+	/**
+	 * Protocol to be used for receiving mail.
+	 */
+	private String protocol;
+
+	/**
+	 * Username for email account. Usually is email id
+	 */
+	private String username;
+
+	/**
+	 * Password for email account.
+	 */
+	private String password;
+
+	/**
+	 * Host for email access
+	 */
+	private String host;
+
+	/**
+	 * Port to be used for email e.g. 993 for imaps
+	 */
+	private String port;
+
+	/**
+	 * Email folder to be scanned for email
+	 */
+	private String folder;
+
+	/**
+	 * Set to true to mark email as read.
+	 */
+	private boolean markAsRead = false;
+
+	/**
+	 * Set to true to delete email after download.
+	 */
+	private boolean delete = false;
+
+	/**
+	 * Set to true to use IdleImap Configuration.
+	 */
+	private boolean idleImap = false;
+
+	/**
+	 * JavaMail properties as a new line delimited string of name-value pairs, e.g. 'foo=bar\n baz=car'.
+	 */
+	private Properties javaMailProperties;
```

Comment:
I'm afraid but this one can't be `null`.
Therefore some default to just `new Properties();`.